front/pages/api/login.ts

```diff
@@ -56,6 +62,32 @@ async function createWorkspace(session: any) {
   return workspace;
 }
 
+async function findWorkspaceWithWhitelistedDomain(session: any) {
+  const { user } = session;
+
+  if (!isGoogleSession(session) || !user.email_verified) {
+    return;
```

Comment:
nit: this caused a parse issue for me

IMHO it's worth distinguishing functions that should return something with functions that shouldn't (void)

Here, the function should return something, possibly indicating we didn't find anything

=> either return undefined (even if it's the same than return without anything, it's clearer) or return null which is kinda type-safer in a figurative way